File: ebs/lib/model/Designation.dart
class Designation {
  final int? id;
  final int? designationCode;
  final String? designationName;
  final int? level;

  Designation({
    this.id,
    this.designationCode,
    this.designationName,
    this.level,
  });

  factory Designation.fromJson(Map<String, dynamic> json) {
    return Designation(
      id: json['id'],
      designationCode: json['designationCode'],
      designationName: json['designationName'],
      level: json['level'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'designationCode': designationCode,
      'designationName': designationName,
      'level': level,
    };
  }
}

File: ebs-springboot/src/main/java/com/erp/model/Designation.java
package com.erp.model;

import javax.persistence.*;

@Entity(name = "designation")
@Table(name = "designation")
public class Designation {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;

    @Column(name = "designation_code")
    private Long designationCode;

    @Column(name = "designation_name")
    private String designationName;

    @Column(name = "level")
    private Integer level;

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public Long getDesignationCode() {
        return designationCode;
    }

    public void setDesignationCode(Long designationCode) {
        this.designationCode = designationCode;
    }

    public String getDesignationName() {
        return designationName;
    }

    public void setDesignationName(String designationName) {
        this.designationName = designationName;
    }

    public Integer getLevel() {
        return level;
    }

    public void setLevel(Integer level) {
        this.level = level;
    }
    
}
